src/syntaxtree.c

```diff
@@ -107,16 +128,22 @@ static Obj SyntaxTreeArgcompInt(UInt i)
     return INTOBJ_INT(i);
 }
 
+static Expr SyntaxTreeArgCodeInt(Obj i)
+{
+    return INT_INTOBJ(i);
+}
+
 static Obj SyntaxTreeRNam(Expr expr)
 {
     return NAME_RNAM(expr);
 }
 
-static Obj SyntaxTreeRefLVar(Obj result, Expr expr)
+static Expr SyntaxTreeCodeRNam(Obj rnam)
 {
-    return INTOBJ_INT(LVAR_REFLVAR(expr));
+    return RNamName(CSTR_STRING(rnam));
```

Comment:
Better (and with input validation):
```suggestion
    return RNamObj(rnam);
```
But I actually would get rid of `SyntaxTreeCodeRNam` and replace it by `RNamObj` directly. And replace `SyntaxTreeRNam` by `NAME_RNAM`.